consensus/api/src/conversions.rs

```diff
@@ -870,6 +812,99 @@ impl TryInto<TransactionValidationError> for ProposeTxResult {
     }
 }
 
+/// Convert transaction::Block --> blockchain::Block.
+impl From<&transaction::Block> for blockchain::Block {
+    fn from(other: &transaction::Block) -> Self {
+        let mut block = blockchain::Block::new();
+        block.set_id(other.id.as_ref().to_vec());
+        block.set_version(other.version);
+        block.set_parent_id(other.parent_id.as_ref().to_vec());
+        block.set_index(other.index);
+        block.set_root_element((&other.root_element).into());
+        block.set_contents_hash(other.contents_hash.as_ref().to_vec());
+        block
+    }
+}
+
+/// Convert blockchain::Block --> transaction::Block.
+impl TryFrom<&blockchain::Block> for transaction::Block {
+    type Error = ConversionError;
+
+    fn try_from(value: &blockchain::Block) -> Result<Self, Self::Error> {
+        let block_id = transaction::BlockID::try_from(value.id.as_slice())?;
+        let parent_id = transaction::BlockID::try_from(value.parent_id.as_slice())?;
+        let root_element = TxOutMembershipElement::try_from(value.get_root_element())?;
+        let contents_hash =
+            transaction::BlockContentsHash::try_from(value.contents_hash.as_slice())?;
+
+        let block = transaction::Block {
+            id: block_id,
+            version: value.version,
+            parent_id,
+            index: value.index,
+            root_element,
+            contents_hash,
+        };
+        Ok(block)
+    }
+}
+
+impl From<&transaction::BlockContents> for blockchain::BlockContents {
+    fn from(source: &transaction::BlockContents) -> Self {
+        let mut block_contents = blockchain::BlockContents::new();
+
+        let key_images: Vec<external::KeyImage> = source
+            .key_images
+            .iter()
+            .map(external::KeyImage::from)
+            .collect();
+
+        let outputs = source.outputs.iter().map(external::TxOut::from).collect();
+
+        block_contents.set_key_images(RepeatedField::from_vec(key_images));
+        block_contents.set_outputs(outputs);
```

Comment:
Curious why one call needs `RepeatedField` and the other doesn't?

Reply:
Good catch